src/error-handler.h

```diff
@@ -24,14 +24,14 @@
 
 namespace wabt {
 
-class SourceErrorHandler {
+class ErrorHandler {
  public:
-  explicit SourceErrorHandler(Location::Type);
+  explicit ErrorHandler(Location::Type);
 
-  virtual ~SourceErrorHandler() {}
+  virtual ~ErrorHandler() {}
 
   // Returns true if the error was handled.
-  virtual bool OnError(const Location*,
+  virtual bool OnError(const Location&,
                        const std::string& error,
                        const std::string& source_line,
                        size_t source_line_column_offset) = 0;
```

Comment:
Why not introduce an inline version without source line and column, for use with the binary case. I.e. something like:

bool OnError(Offset offset, const std::string& error) {
  return OnError(Location(offset), error, std::string(), 0);
}

That way, the binary use doesn't need to explicitly convert/fill in arguments.